### PR TITLE
STCOM-842 re-scale cached sizes to fit...

### PR DESCRIPTION
**Pull request by JohnC-80**
## Purpose
This PR resolves issue [STCOM-842](https://issues.folio.org/browse/STCOM-842) where panesets do not fit the window (either they're too small leaving whitespace or the panes are cropped off.) If a user resizes the panes, travels to a different app, resizes the window, then returns to the app where they resized, the old pane dimensions remain which will not match the resized window (*Phew!)

### Approach
a) use the resize container (a full width div) to judge the real width of the window for the top-level paneset.
b) if an initial layout prop is provided, rescale it before applying it to the panes. - sum the widths, compare to the resize container, rescale the values accordingly...

Math. Wooo 🎏

### CODE REVIEW

File: lib/Paneset/Paneset.js [hunk 1]
```diff
@@ -311,13 +355,18 @@ class Paneset extends React.Component {
...
+        const adjustedWidths = this.resizeCachedSizesToFit(sizesCached);
+        widths = adjustedWidths || widths;
+        this.updateLayoutCache(widths, changeType);
+        this.resizePanes(this.state.panes, widths);
       } else {
         widths = this.calcWidths(this.state.panes);
         this.updateLayoutCache(widths, changeType);
+        this.resizePanes(this.state.panes, widths);
       }
-      this.resizePanes(this.state.panes, widths);
```
Comment (NikitaSedyx) [inline lib/Paneset/Paneset.js:320]: looks like it can be unchanged

Comment (NikitaSedyx) [reply lib/Paneset/Paneset.js:320]: and now `this.updateLayoutCache(widths, changeType);` is called in all if-else blocks

File: lib/Paneset/Paneset.js [hunk 2]
```diff
@@ -311,13 +355,18 @@ class Paneset extends React.Component {
       if (toApply && sizesCached === -1) {
         widths = toApply;
         this.updateLayoutCache(widths, changeType);
+        this.resizePanes(this.state.panes, widths);
       } else if (sizesCached !== -1) {
         widths = this.state.layoutCache[sizesCached];
+        const adjustedWidths = this.resizeCachedSizesToFit(sizesCached);
+        widths = adjustedWidths || widths;
```
Comment (NikitaSedyx) [inline lib/Paneset/Paneset.js:362]: 3 lines could be `widths = this.resizeCachedSizesToFit(sizesCached) || this.state.layoutCache[sizesCached];`

**Review by NikitaSedyx — APPROVED** [review]
nice, thank you!

File: lib/Paneset/Paneset.js (hunk 2 above)
Comment (JohnC-80) [reply lib/Paneset/Paneset.js]: Thank you! all DRYed up...

File: lib/Paneset/Paneset.js [hunk 3]
```diff
@@ -14,6 +14,14 @@ const defaultProps = {
   initialLayouts: [],
 };
 
+function sumWidths(key, objectArray) {
+  let sum = 0;
+  objectArray.forEach(o => {
+    sum += parseInt(o[key], 10);
```
Comment (zburke) [inline lib/Paneset/Paneset.js:20]: Maybe `reduce` is more idiomatic? Dunno.
```
return objectArray.reduce((a, i) => parseInt(i[key], 10) + a, 0);
```

File: lib/Paneset/Paneset.js (hunk 1 above)
Comment (zburke) [reply lib/Paneset/Paneset.js:320]: Right, so the question is, which is correct?
* the previous impl when it _wasn't_ called given `sizesCached >= 0`
* the current impl when it is _always_ called, regardless of the value of `sizesCached`

**Pull request merged**